docs/sql-ref-datatypes.md

```diff
@@ -49,6 +49,33 @@ Spark SQL and DataFrames support the following data types:
   absolute point in time.
   - `DateType`: Represents values comprising values of fields year, month and day, without a
   time-zone.
+* Interval types
```

Comment:
nit: Can we add a description: "individual interval fields are non-negative but interval itself can have a sign, and be negative."